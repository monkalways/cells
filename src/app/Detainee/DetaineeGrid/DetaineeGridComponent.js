import React from 'react';
import PropTypes from 'prop-types';
import { Grid, withStyles } from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export const DetaineeGridComponent = ({ classes, children }) => (
  <Grid container className={classes.container} spacing={8}>
    {children}
  </Grid>
);

DetaineeGridComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  container: {
    backgroundColor: theme.palette.background.default,
    height: theme.spacing.unit * 113,
    marginLeft: 0,
    marginRight: 0,
    marginBottom: theme.spacing.unit * 0.4,
    padding: theme.spacing.unit,
    width: '100%',
  },
}))(DetaineeGridComponent);
