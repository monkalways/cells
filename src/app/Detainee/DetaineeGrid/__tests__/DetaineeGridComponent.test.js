import React from 'react';
import { shallow } from 'enzyme';

import { DetaineeGridComponent } from '../DetaineeGridComponent';

describe('DetaineeGridComponent', () => {
  let children;

  beforeEach(() => {
    children = <h1>Hello Test</h1>;
  });

  const setup = () => {
    const classes = {
      container: 'container',
    };

    return shallow(<DetaineeGridComponent classes={classes} children={children} />);
  };

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
