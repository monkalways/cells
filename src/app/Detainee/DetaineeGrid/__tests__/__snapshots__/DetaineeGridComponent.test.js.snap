// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetaineeGridComponent should render 1`] = `
<WithStyles(Grid)
  className="container"
  container={true}
  spacing={8}
>
  <h1>
    Hello Test
  </h1>
</WithStyles(Grid)>
`;
